app/statistics/controller.js

```diff
@@ -266,15 +190,63 @@ export default class StatisticsController extends Controller.extend(
     });
   }
 
+  @action
+  changeFromDate(date) {
+    this.fromDate = serializeMoment(date);
+    this.data.perform();
+  }
+
+  get getFromDate() {
+    return deserializeMoment(this.fromDate);
+  }
+
+  @action
+  changeToDate(date) {
+    this.toDate = serializeMoment(date);
+    this.data.perform();
+  }
+
+  get getToDate() {
+    return deserializeMoment(this.toDate);
+  }
+
   @action
   setModelFilter(key, value) {
     set(this, key, value && value.id);
+    this.data.perform();
   }
 
   @action
   resetQP() {
     this.resetQueryParams(
-      Object.keys(this.allQueryParams).filter((qp) => qp !== "type")
+      Object.keys(allQueryParams(this)).filter((qp) => qp !== "type")
     );
   }
+
+  get queryParamsState() {
+    const states = {};
+    for (const param of this.queryParams) {
+      const defaultValue = getDefaultQueryParamValue(param);
+      const currentValue = this[param];
+      states[param] = {
+        as: param,
+        defaultValue,
+        serializedValue: currentValue,
+        value: currentValue,
+        changed: currentValue !== defaultValue,
+      };
+      if (["fromDate", "toDate"].includes(param)) {
+        states[param].serialize = serializeMoment;
+        states[param].deserialize = deserializeMoment;
+      }
+    }
+    return states;
+  }
```

Comment:
Export the contents of this getter to the `utils/query-param.js` file as well and maybe give it a `dates` parameter so you can dynamically adapt the *qp*s which need date formatting.